algorithms/src/Kokkos_Random.hpp

```diff
@@ -902,16 +906,20 @@ class Random_XorShift64_Pool {
     locks_      = src.locks_;
     state_      = src.state_;
     num_states_ = src.num_states_;
+    padding_    = src.padding_;
     return *this;
   }
 
   void init(uint64_t seed, int num_states) {
     if (seed == 0) seed = uint64_t(1318319);
 
+    padding_    = num_states < 1000 ? 64 : 1;
```

Comment:
No magic numbers.  Add a comment.

Reply:
I added a comment that this is a magic number :-P

Reply:
seriously I just needed something not too small and not too big 16 or 32 would probably have worked too.

Reply:
https://github.com/microsoft/STL/blob/9a9820df1a1d3fa84100e3169ff37fdd4fa41759/stl/inc/new#L40

Reply:
That is, `hardware_destructive_interference_size` perhaps?

Reply:
What about the `1000`